lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt

```diff
@@ -0,0 +1,44 @@
+package org.partiql.lang.eval.visitors
+
+import org.partiql.lang.domains.extractSourceLocation
+
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.eval.extractColumnAlias
+import org.partiql.pig.runtime.SymbolPrimitive
+
+/**
+ * Assigns aliases to any unspecified [PartiqlAst.FromSource.Scan]/[PartiqlAst.FromSource.Unpivot] that does not
+ * already have one.
+ *
+ * For example: `SELECT * FROM foo` gets rewritten as `SELECT * from foo as foo`.
+ * Path expressions:  `SELECT * FROM foo.bar.bat` gets rewritten as `SELECT * from foo.bar.bat as bat`
+ *
+ * If provided with a query that has all of the from source aliases already specified, an exact clone is returned.
+ */
+class FromSourceAliasVisitorTransform : PartiqlAst.VisitorTransform() {
+
+    private class InnerFromSourceAliasVisitorTransform : PartiqlAst.VisitorTransform() {
+        private var fromSourceCounter = 0
+
+        override fun transformFromSourceScan_asAlias(node: PartiqlAst.FromSource.Scan): SymbolPrimitive? {
+            val thisFromSourceIndex = fromSourceCounter++
+            return node.asAlias ?:
+                transformSymbolPrimitive(
+                    SymbolPrimitive(node.expr.extractColumnAlias(thisFromSourceIndex), node.extractSourceLocation()))
```

Comment:
I don't believe it is necessary to transform the newly constructed `SymbolPrimitive` so I think you can do away with the call to `transformSymbolPrimitive`.

Reply:
Got it, removed.